Dimension errors in tests of duck_db_adapter -> search

### Description

In #56 and commit 0f49a9fcd7290f1c207c0e4f2c30ee6f286f613f we handled the issue of using the correct vector dimensions when using search from the CLI. However it still occurs that when the model is None and defaults to the `DEFAULT_MODEL` (sentence-transformer model) it cannot fetch its dimension due to wrong conditional statements. 

The error is exactly the same:
```
E       duckdb.duckdb.ConversionException: Conversion Error: Cannot cast array of size 384 to array of size 1536
E       LINE 2: ...     SELECT *, array_distance(embeddings::FLOAT[1536],
```

### Solution:
Add conditional statement when model_name string starts with `"all-MiniLM-L6-v2"`, return dimension for `DEFAULT_MODEL`. By [now](https://github.com/monarch-initiative/curate-gpt/blob/32fa049e4d716a835b761a3b8ca16720b0c2f9e5/src/curate_gpt/store/duckdb_adapter.py#L1022) it would return the `DEFAULT_OPENAI_MODEL` as it is not None but also does not start with `"openai:"`

--- a/src/curate_gpt/store/duckdb_adapter.py
+++ b/src/curate_gpt/store/duckdb_adapter.py
@@ -1020,10 +1020,9 @@ class DuckDBAdapter(DBAdapter):
         return " AND ".join(conditions)
 
     def _get_embedding_dimension(self, model_name: str) -> int:
-        if model_name is None:
+        if model_name is None or model_name.startswith(self.default_model):
             return DEFAULT_MODEL[self.default_model]
         if isinstance(model_name, str):
-            logger.info("somehow here")
             if model_name.startswith("openai:"):
                 model_key = model_name.split("openai:", 1)[1]
                 model_info = MODEL_MAP.get(model_key, DEFAULT_OPENAI_MODEL)

